src/main/java/io/cdap/plugin/MultiTableDBSource.java

```diff
@@ -73,19 +87,72 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   public void prepareRun(BatchSourceContext context) throws Exception {
     Configuration hConf = new Configuration();
     Class<? extends Driver> driverClass = context.loadPluginClass(JDBC_PLUGIN_ID);
-    Collection<DBTableInfo> tables = MultiTableDBInputFormat.setInput(hConf, conf, driverClass);
-    SettableArguments arguments = context.getArguments();
-    for (DBTableInfo tableInfo : tables) {
-      arguments.set(DynamicMultiFilesetSink.TABLE_PREFIX + tableInfo.getDbTableName().getTable(),
-                    tableInfo.getSchema().toString());
-    }
 
-    context.setInput(Input.of(conf.getReferenceName(),
-                              new SourceInputFormatProvider(MultiTableDBInputFormat.class, hConf)));
+    Collection<DBTableInfo> tables;
+
+    try {
+      tables = MultiTableDBInputFormat.setInput(hConf, conf, driverClass);
+
+      SettableArguments arguments = context.getArguments();
+      for (DBTableInfo tableInfo : tables) {
```

Comment:
this argument setting is actually what we want to remove (along with making the sink not require it). If we remove this, there won't be the possibility of an exception being thrown, so no need for the InputFormat that always emits errors.

The argument to keep this would be to make it compatible with older multi-sink plugins, but it wouldn't really be compatible anyway for the new setting where the user passes in sql queries, since there isn't an efficient way to get the table name and schema from just the query.

Reply:
I'm gonna put this behind a configuration setting (let's say `Populate Schema`). If enabled, we set the schema up. Otherwise, we leave it blank (and avoid this potential problem).

By default, we should turn it off, but customers can turn it on for backwards compatibility purposes if they so desire.

How does this sound?

Reply:
that sounds fine, though for compatibility it should probably default to on then.